app/client/components/homePage/banner/Countdown.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { CSSProperties, useEffect, useState } from "react";
+
+interface CountdownProps {
+  endDate: Date;
+  onCountdownFinish: () => void;
+}
+
+interface CountdownSpanCSS extends CSSProperties {
+  "--value": number;
+}
+
+interface TimeLeft {
+  days: number;
+  hours: number;
+  minutes: number;
+  seconds: number;
+}
+
+const calculateTimeLeft = (endDate: Date): TimeLeft => {
```

Comment:
Maybe you could add an external library in order to handle time operations and parsing, e.g. `date-fns` or `moment.js`?

Reply:
I thought about it, but I prefer to keep it simple and stick to built-in `Date` and `Intl` objects. 

`moment.js` is outdated and authors don't recommend it anymore in [official docs](https://momentjs.com/docs/#/-project-status/) 

`date-fns` looks interesting, but I don't think we need most of the features it offers. But if it later turns out that such a library will make life easier, then I will switch to it.